pkg/api/v2/reporter/reporter.go

```diff
@@ -128,6 +131,21 @@ func (e ResourceReports) AddWarning(res resources.InputResource, warning string)
 	e[res] = rpt
 }
 
+func (e ResourceReports) AddMessages(res resources.InputResource, message ...string) {
+	for _, warn := range message {
+		e.AddMessages(res, warn)
+	}
+}
+
+func (e ResourceReports) AddMessage(res resources.InputResource, message string) {
+	if message == "" {
+		return
+	}
+	rpt := e[res]
```

Comment:
check if not found, and return if `res` is not in the map

Reply:
But then we wouldnt be able to create new messages, only modify existing ones wouldnt we?

Reply:
what happens if `e[res]` returns `nil, false`? i.e. it's not in the map.. then on line 156 I think this will panic since `rpt.Messages` will dereference a nil pointer. Let's add a unit test to prove this out / catch this